File: 1931.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Comparator;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int N = Integer.parseInt(br.readLine()); // 회의의 수
        int[][] time = new int[N][2]; // 시작 시간, 종료 시간

        for(int i = 0; i < N; i++){ // 회의 시간 N번 입력
             st = new StringTokenizer(br.readLine()," ");
             time[i][0] = Integer.parseInt(st.nextToken()); // 시작 시간
             time[i][1] = Integer.parseInt(st.nextToken()); // 종료 시간
        }

        Arrays.sort(time, new Comparator<int[]>() {
            @Override
            public int compare(int[] o1, int[] o2) {
                if(o1[1] == o2[1]){ // 종료 시간이 같을 경우 시작 시간이 빠른순으로 정렬한다.
                    return o1[0] - o2[0];
                }
                return o1[1] - o2[1];
            }
        });

        int count = 0; // 회의 개수
        int end_time = 0;

        for(int i = 0; i < N; i++){
            if(end_time <= time[i][0]){
                count++;
                end_time = time[i][1];
            }
        }

        System.out.println(count);
    }
}

File: python/1931.py
import sys

# 입력
n = int(sys.stdin.readline())  # 회의의 수 N

for _ in range(n): # 회의 시간 입력
    start, end = map(int, input().split())
    time.append([start, end])

time = []  # 회의 시간
count = 1  # 회의의 개수

time.sort(key=lambda x: (x[1], x[0])) # 시작시간, 종료시간을 기준으로 회의 시간 정렬

end_time = time[0][1] # 첫 회의 종료 시간
for i in range(1, n):
    if end_time <= time[i][0]: # 회의 종료 시간과 같거나 큰 회의 시작 시간이 있다면
        end_time = time[i][1] # 회의 종료 시간 갱신
        count += 1 # 최대 회의 개수 증가

print(count)
